src/components/customSearch/AddFieldOrGroupButton.tsx

```diff
@@ -0,0 +1,29 @@
+import AddIcon from '@mui/icons-material/Add'
+import { Button } from '@mui/material'
+import { SearchParameter } from '../../types/SimpleSearch'
+
+interface AddFieldOrGroupButtonProps {
+  onClick: (obj: any) => void
+  initialCustomSearchValues: SearchParameter
+  label: string
+}
+const AddFieldOrGroupButton: React.FC<any> = ({
+  onClick,
+  initialCustomSearchValues,
+  label
+}) => {
+  return (
+      <Button
+        variant="text"
+        startIcon={<AddIcon />}
+        onClick={() => {
+          onClick(initialCustomSearchValues)
```

Comment:
I think we don't need to pass "initialCustomSearchValues" as a prop and then sent it back on click ?
Should rather be `onClick={onClick}` and then handle that in the parent component.